server/webapp/WEB-INF/rails.new/webpack/models/metrics_consent/metrics_settings.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const Routes     = require('gen/js-routes');
+const AjaxHelper = require('helpers/ajax_helper');
+const Stream     = require('mithril/stream');
+
+//todo: Blame Ganeshpl for this
+Routes.apiv1ShowMetricsSettingsPath  = () => {
+  return '/go/api/metrics/settings';
```

Comment:
Metrics Settings path doesnt come from `gen/js-routes`.
Define `/go/api/metrics/settings` path explicitly in `helpers/spark_routes`